### ISSUE TITLE
Fails to connect to (SQLite) DB when exporting a function within a Migration file

### ISSUE THREAD

**Issue by persunde** [typeorm/typeorm#7836]
## Issue Description
Typeorm fails to connect to the SQLite database when a function is exported within a file in the migration folder. This happens when you just try to run the program normally and not running any migrations.

I have the same issue as mentioned here: https://github.com/typeorm/typeorm/issues/5458
I also have a different Error so I created this bug report.

I fixed the issue by removing the "export" from any function I had in a file in the MIgration folder, as mentioned here:
https://github.com/typeorm/typeorm/issues/5458#issuecomment-751883844
But it took me a long time to fix the issue, since the error message I received made no sense to me.

### Expected Behavior
I expected Typeorm to be able to connect to the Database. Or give an accurate error message.

### Actual Behavior
The programs throws a `ConnectionIsNotSetError` or `TypeError: someClass is not a constructor` and is unable to connect to the SQLite database when running the program normally, aka `yarn start`.
It also happens when I run the typeorm (TS) migration `yarn typeorm migration:run` (see my package.json file further down).

Here are the two different error messages I get:

```
ConnectionIsNotSetError: Connection with sqlite database is not established. Check connection configuration.
    at new ConnectionIsNotSetError (.../src/error/ConnectionIsNotSetError.ts:8:9)
    at SqliteQueryRunner.query (.../src/driver/sqlite/SqliteQueryRunner.ts:44:19)
    at Connection.<anonymous> (.../src/connection/Connection.ts:404:42)
    at step (.../node_modules/tslib/tslib.js:141:27)
    at Object.next (.../node_modules/tslib/tslib.js:122:57)
    at .../node_modules/tslib/tslib.js:115:75
    at new Promise (<anonymous>)
    at Object.__awaiter (.../node_modules/tslib/tslib.js:111:16)
    at Connection.query (.../node_modules/typeorm/connection/Connection.js:402:24)
    at Object.getImagesForMigration (.../src/testscripts/image_script.ts:8:6)
```

```
TypeError: someClass is not a constructor
    at class_1.get (.../src/container.ts:34:51)
    at Object.getFromContainer (.../src/container.ts:71:29)
    at .../src/connection/ConnectionMetadataBuilder.ts:35:58
    at Array.map (<anonymous>)
    at ConnectionMetadataBuilder.buildMigrations (.../src/connection/ConnectionMetadataBuilder.ts:35:36)
    at Connection.buildMetadatas (.../src/connection/Connection.ts:521:54)
    at Connection.<anonymous> (.../src/connection/Connection.ts:193:18)
    at step (.../node_modules/tslib/tslib.js:141:27)
    at Object.next (.../node_modules/tslib/tslib.js:122:57)
    at fulfilled (.../node_modules/tslib/tslib.js:112:62)
```

### Steps to Reproduce
Export a function from a file located in the Migration folder. 
Here is a file I used to replicate the problem. 

1. Copy the code below into a file and put the file under the Typeorm Migration Folder

```typescript
import {MigrationInterface, QueryRunner} from "typeorm";

export class JobPostImageMigration1624873168865 implements MigrationInterface {
    name = 'JobPostImageMigration1624873168865'

    public async up(queryRunner: QueryRunner): Promise<void> {
        getImagesForMigration()
    }

    public async down(queryRunner: QueryRunner): Promise<void> {
    }

}

export async function getImagesForMigration() {}
```

2. Run the code below with `npm run start` or `yarn start`
My start script for package.json looks like this:
```
"scripts": {
      "start": "ts-node src/index.ts",
      "typeorm": "node --require ts-node/register ./node_modules/typeorm/cli.js"
}
```

```typescript
import { createConnection, getConnection } from "typeorm"

getErrorMessage()


async function getDB() {
  let connectionPool = await createConnection() 
	
	const connection = getConnection();
	return connection
}

async function getErrorMessage() {
	const con = await getDB()

  const result = await con
    .query(`
    SELECT *
    FROM test_table
  `);
}
```

3. The program will exit with an error `TypeError: someClass is not a constructor`
I am unable to create a small file that throws the `ConnectionIsNotSetError: Connection with sqlite database is not established. Check connection configuration.` error, but it happens for the same reason.

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |  Ubuntu 18.04.5 LTS  |
| Node.js version     | v14.15.1 | 
| Typescript version  | v4.1.3 |
| TypeORM version     | v0.2.31 |

### Additional Context

The error causes trouble for running the program normally as well as for running migrations. But the errors are different.
Typeorm creates the SQLite Database file, but then throws an error.

### Relevant Database Driver(s)

- [ ] `aurora-data-api`
- [ ] `aurora-data-api-pg`
- [ ] `better-sqlite3`
- [ ] `cockroachdb`
- [ ] `cordova`
- [ ] `expo`
- [ ] `mongodb`
- [ ] `mysql`
- [ ] `nativescript`
- [ ] `oracle`
- [ ] `postgres`
- [ ] `react-native`
- [ ] `sap`
- [x] `sqlite`
- [ ] `sqlite-abstract`
- [ ] `sqljs`
- [ ] `sqlserver`

### Are you willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time, and I know how to start.
- [x] Yes, I have the time, but I don't know how to start. I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by imnotjames** [comment]
You currently cannot export anything from migration files except for migrations if you're using the file based loading.

**Comment by imnotjames** [comment]
As this is indeed a duplicate of #5458 I'll be closing this one

**Comment by persunde** [comment]
> You currently cannot export anything from migration

I know. It was just a mistake on my side as I copied the code from a different part of my code base where this function was exported.

This bug caused me a lot of lost time, as I could not understand why the program crashed. It was very hard to connect the dots that this bug caused my program to crash when I just wanted to run the program normally and not even run the Migration file.

When I encountered this bug `ConnectionIsNotSetError: Connection with sqlite database is not established. Check connection configuration.` I could not find any helpful info online.
Only when I encountered the second bug `TypeError: someClass is not a constructor` did I find helpful information in one of the bug reports.

I hope this bug will be fixed soon :)
